bullet_train-super_scaffolding/lib/bullet_train/super_scaffolding/scaffolders/crud_scaffolder.rb

```diff
@@ -35,6 +35,20 @@ def run
           parents = parents.map(&:classify).uniq
           parent = parents.first
 
+          migration_file_name = `grep "create_table :#{child.tableize}" db/migrate/*`.split(":").shift
+          parent_foreign_key = nil
+          File.open(migration_file_name).readlines.each do |line|
+            parent_foreign_key = line.match?("t.references :#{parent.tableize.singularize}")
+            break if parent_foreign_key
+          end
```

Comment:
I think we might want to alter this block somehow so that we don't assume that the needed reference was added at the time the table was created.

Like what if I did:

```
rails g model Project name:title
rails db:migrate
rails g migration AddTeamToProjects team:references
rails db:migrate
```

The column would be there, but this block would think it was missing and would refuse to generate a scaffold. (I have no idea how common the scenario I described might be.)

Reply:
Oh, I just remembered that we tell people not to run migrations after generating their model. https://bullettrain.co/docs/super-scaffolding#1-basic-crud-scaffolding-with-crud

So maybe the concern I mentioned above is mostly invalid?

Should we add a flag to allow someone to say `--skip-parent-reference-check` or something?

Reply:
@jagthedrummer

> Oh, I just remembered that we tell people not to run migrations after generating their model. https://bullettrain.co/docs/super-scaffolding#1-basic-crud-scaffolding-with-crud

Yes, this is mainly what I had in mind. I primarily wanted to do something like `ModelName.column_names` to check for the reference, but since we don't run the migrations before Super Scaffolding I wasn't able to grab the reference that way.

However, I think your point is still valid. We could technically do this:
```
rails g model Project name:title
rails g migration AddTeamToProjects team:references
rails db:migrate
```

What I could do is check for the add_column reference like we do in this code for boolean buttons (we had a bug revolving around the same issue in #395):

https://github.com/bullet-train-co/bullet_train-core/blob/eb1758bdfedefdbdf41c93e009de952954a65d1c/bullet_train-super_scaffolding/lib/scaffolding/transformer.rb#L1161-L1165

I'm not sure of any other scenarios that could cause trouble for the parent foreign key check, but I think we'll be safe if we at least add this part.

Reply:
@jagthedrummer I made some new editions, it currently looks like this. I decided to use `puts` instead of `raise` since it's easier to look at.

![image](https://github.com/bullet-train-co/bullet_train-core/assets/10546292/a20482c3-fd28-4187-905b-4a8e51af4a98)